PanoptesNetClient/PanoptesNetClient/Request.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Newtonsoft.Json.Linq;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace PanoptesNetClient
+{
+    public class Request : IRequest
+    {
+        private string Resource;
+        private string Id;
+        public string Endpoint { get; set; }
+        private string Query;
+
+        public Request(string resource)
+        {
+            Resource = resource;
+            Endpoint = $"api/{resource}/";
+        }
+
+        public IRequest ById(string id)
+        {
+            Id = id;
+            return this;
+        }
+
+        public IRequest WithArgs(NameValueCollection query)
+        {
+            var collection = HttpUtility.ParseQueryString(string.Empty);
+
+            foreach (var key in query.Cast<string>().Where(key => !string.IsNullOrEmpty(query[key])))
+            {
+                collection[key] = query[key];
+            }
+            Query = $"?{collection.ToString()}";
+            return this;
+        }
+
+        public async Task<JObject> GetAsync()
```

Comment:
Seems kind of weird that the `Request` is both storing information about the request and actually making the call to the `ApiClient`. The way I imagined this working is that a request would just hold the information the `ApiClient` would need to make the request and some other piece of code would actually pass it in.

Reply:
Oh, I think I was misunderstanding the architecture here. The `IRequest`s are created by the `ApiClient`